app/services/induction/amend_participant_cohort.rb

```diff
@@ -177,6 +199,16 @@ def target_school_cohort
     end
 
     # Validations
+    def eligible_to_be_moved_after_payments_frozen
+      unless participant_profile.eligible_to_change_cohort_and_continue_training?
+        errors.add(:participant_profile, :participant_not_eligible_to_be_moved)
+      end
+    end
+
+    def target_cohort_payments_active
+      errors.add(:target_cohort_start_year, :payments_frozen) if target_cohort&.payments_frozen?
```

Comment:
I think this needs to be more restrictive; after talking with Nathan it seems we only want to let 2021 participants migrate to 2024. This will check the cohort they are migrating to allows payments, but that may be error prone depending on how this service is called/by who. Similarly, once a participant has migrated to 2024 we may want to migrate them _back_ to 2021 (if it was done by mistake). We should be able to do this providing the participant didn't get any billable declarations while in 2024 (but if they have billable declarations in 2021 and they span multiple cohorts they should still be able to migrate back to 2021).

In the `ChangeSchedule` service I'm going to apply this logic:

- When transferring from a frozen cohort you can only transfer into a cohort that is 3 years ahead (so 2021 -> 2024).
- When transferring to an earlier cohort and the participant has `cohort_changed_after_payments_frozen = true`, the cohort must be 3 years earlier and have payments frozen (so 2024 -> 2021). The participant must also have no billable declarations in their current cohort (2024).

Reply:
That validation is to prevent participants to be sitting in payment-frozen cohorts.
Given that Nathan allows 2021 participants to go to 2024 and back, I will accomodate it

Reply:
The 3 years rule could be too restrictive IMO. What if a 2021 participant resumes training not this year but next year?
This service only allows cohort changes of 2021 participants to the cohort currently open for registration if their original cohort gets payments-frozen.